File: easy/3392.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    int countSubarrays(vector<int> &nums)
    {
        int result = 0;
        for (int i = 0; i < nums.size() - 2; i++)
        {
            if (nums[i] + nums[i + 2] == nums[i + 1] / 2.0)
            {
                result++;
            }
        }
        return result;
    }
};

int main()
{
    vector<int> nums = {-1, -4, -1, 4};
    Solution sol;
    cout << sol.countSubarrays(nums);
}

File: easy/3392.py
from typing import List

class Solution:
    def countSubarrays(self, nums: List[int]) -> int:
        result = 0
        for i in range(len(nums)-2):
            if nums[i] + nums[i+2] == nums[i+1]/2:
                result += 1
        return result
    
if __name__ == "__main__":
    nums = [1,2,1,4,1]
    print(Solution().countSubarrays(nums=nums))